src/extensions/sort.rs

```diff
@@ -0,0 +1,148 @@
+//! Adds support for the IMAP SORT extension specificed in [RFC
+//! 5464](https://tools.ietf.org/html/rfc5256#section-3).
+//!
+//! The SORT command is a variant of SEARCH with sorting semantics for
+//! the results. There are two arguments before the searching
+//! criteria argument: a parenthesized list of sort criteria, and the
+//! searching charset.
+
+use std::{borrow::Cow, fmt};
+
+pub(crate) struct SortCriteria<'c>(pub(crate) &'c [SortCriterion<'c>]);
+
+impl<'c> fmt::Display for SortCriteria<'c> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.0.is_empty() {
+            write!(f, "")
+        } else {
+            let criteria: Vec<String> = self.0.iter().map(|c| c.to_string()).collect();
+            write!(f, "({})", criteria.join(" "))
+        }
+    }
+}
+
+/// The defined sort criteria are as follows. Refer to the Formal
+/// Syntax section for the precise syntactic definitions of the
+/// arguments. If the associated [RFC-822](https://tools.ietf.org/html/rfc822)
+/// header for a particular criterion is absent, it is treated as the empty string.
+/// The empty string always collates before non-empty strings.
```

Comment:
This text reflects stuff that we as the protocol implementors need to know, but it's not really something the user of `imap` needs to know given that we've abstracted that away from them. Let's update this to something simpler:
```suggestion
/// Message sorting preferences used for [`Session::sort`] and [`Session::uid_sort`].
```

Reply:
:heavy_check_mark: Done